i6_models/parts/fsa.py

```diff
@@ -0,0 +1,144 @@
+from __future__ import annotations
+
+__all__ = ["TorchFsaBuilder", "WeightedFsa"]
+
+from functools import reduce
+from typing import Iterable, NamedTuple, Tuple, Union
+
+import numpy as np
+import torch
+import librasr
+
+
+class WeightedFsa(NamedTuple):
+    """
+    Convenience class that represents an FSA. It supports scaling the weights of the
+    fsa by simple left-multiplication and moving the tensors to a different device.
+    It can simply be passed to :func:`i6_native_ops.fbw.fbw_loss` and :func:`i6_native_ops.fast_viterbi.align_viterbi`.
+
+    :param num_states: the total number of all states S
+    :param edges: a [4, E] tensor of edges with number of edges E and where each column is an edge
+        consisting of from-state, to-state, emission idx and the index of the sequence it belongs to
+    :param weights: a [E,] tensor of weights for each edge scaled by the tdp_scale
+    :param start_end_states: a [N, 2] tensor of start and end states for each of the N sequences
+    """
+
+    num_states: torch.IntTensor
+    edges: torch.IntTensor
+    weights: torch.FloatTensor
+    start_end_states: torch.IntTensor
+
+    def __mul__(self, scale: float) -> WeightedFsa:
+        """Multiply the weights, i.e. the third element, with a scale."""
+        return WeightedFsa(
+            self.num_states,
+            self.edges,
+            self.weights * scale,
+            self.start_end_states,
+        )
+
+    def to(self, device: Union[str, torch.device]) -> WeightedFsa:
+        """Move the tensors to a given device. This wraps around the
+        PyTorch `Tensor.to(device)` method."""
+        return WeightedFsa._make(tensor.to(device) for tensor in self)
+
+
+class TorchFsaBuilder:
```

Comment:
I think everything in i6_models is torch related. The specialty of this class is that it uses rasr as a source for the fsas.
Would then 
```suggestion
class RasrFsaBuilder:
```
be a more appropriate branding for this class here?

Also, would it make sense to rename `fsa.py` to `rasr_fsa.py` just in case we will have different fsa related functions that should not be in the same scope as the `librasr` import?